programs/zstdcli.c

```diff
@@ -597,6 +598,7 @@ int main(int const argCount, const char* argv[])
     const char* outFileName = NULL;
     const char* outDirName = NULL;
     const char* dictFileName = NULL;
+    const char* diffFromDictFileName = NULL;
```

Comment:
Having 2 different variables `dictFileName` and `diffFromDictFileName` makes it now possible to define them both.
What happens then ?
The behavior should be defined, and documented.

Suggestion : 
- Only one of them should be possible. If both are defined, then, it's either an error, or the latest one takes over. I have a small preference for `error` in this case.
- Having a single variable for both makes it impossible to define both. But it requires another state variable to define what the value means. So there is no reduction in nb of variables. Hence, I have no preference : select whichever feels better to you.

Reply:
Yes, just having one variable is very possible. Will update